### PR TITLE
Add Nginx authenticator plugin to Letsencrypt Puppet module

### PR DESCRIPTION
**Pull request by alexfouche**
#### Pull Request (PR) description

I added automatic installation of the `...-certbot-nginx-...` package when `'plugin' => 'nginx'` is given to define `letsencrypt::certonly`. And i provided an example in Readme.

#### This Pull Request (PR) fixes the following issues

On Centos for example, when a Nginx web server is running, simply asking Certbot to create or renew certificates is not enough. It requires an additional package or plugin,

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,19 @@ letsencrypt::certonly { 'foo':
 }
 ```
 
+#### Nginx authenticator
+
+To request a certificate for $::fqdn with the `certonly` installer and the `nginx` authenticator, temporarily opening firewall for HTTP, with an already running Nginx, and reloading Nginx afterwards:
+
+```puppet
+letsencrypt::certonly { $::fqdn:
```
Comment (bastelfreak) [inline README.md]: please don't use the legacy topscope syntax to access facts. instead you can use the facts hash:
```suggestion
letsencrypt::certonly { $facts['fqdn']:
```

File: manifests/plugin/nginx.pp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# === Parameters:
+#
+# [*manage_package*]
+#   Manage the plugin package.
+# [*package_name*]
+#   The name of the package to install when $manage_package is true.
+#
+class letsencrypt::plugin::nginx (
+  Boolean $manage_package          = $letsencrypt::nginx_manage_package,
+  String $package_name             = $letsencrypt::nginx_package_name,
```
Comment (bastelfreak) [inline manifests/plugin/nginx.pp]: we should enforce the minimal string length here to prohibit empty strings:
```suggestion
  String[1] $package_name          = $letsencrypt::nginx_package_name,
```

**Comment by bastelfreak** [comment]
Hi @alexfouche, thanks for the PR. Please have a look at the inline comments I made and also at the failing travis jobs.

File: manifests/certonly.pp [hunk 3]
```diff
@@ -94,6 +94,22 @@
     $default_args = '--text --agree-tos --non-interactive delete'
   }
 
+  # Plugin requirements
```
Comment (ekohl) [inline manifests/certonly.pp]: I don't like duplicating the plugin selector. I'd prefer an `if $plugin == 'nginx' { require letsencrypt::plugin::nginx }` inside the default block.

File: manifests/certonly.pp [hunk 4]
```diff
@@ -118,7 +133,7 @@
       ]
     }
 
-    default: {
+    'nginx', default: {
```
Comment (ekohl) [inline manifests/certonly.pp]: This is technically redundant.

File: manifests/certonly.pp [hunk 5]
```diff
@@ -66,6 +66,7 @@
   Array[String[1]]                          $environment          = [],
   Boolean                                   $manage_cron          = false,
   Boolean                                   $suppress_cron_output = false,
+  Optional[String]                          $cron_output_mailto   = undef,
```
Comment (ghoneycutt) [inline manifests/certonly.pp:69]: `Optional[String[1]]`

**Comment by ekohl** [comment]
There are merge conflicts now. Any plans to rebase this and address the comments?

**Comment by kenyon** [comment]
Done in #267.

**Pull request closed without merging**